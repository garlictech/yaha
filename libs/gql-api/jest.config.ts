module.exports = {
  displayName: 'gql-api',

  globals: {},
  testEnvironment: 'node',
  transform: {
    '^.+\\.[tj]sx?$': [
      'ts-jest',
      {
        tsconfig: '<rootDir>/tsconfig.spec.json',
      },
    ],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/gql-api',
  preset: '../../jest.preset.ts',
};
