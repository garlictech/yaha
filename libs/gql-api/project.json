{
  "name": "gql-api",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/gql-api/src",
  "projectType": "library",
  "targets": {
    "build-schema": {
      "executor": "@nx/workspace:run-commands",
      "options": {
        "parallel": false,
        "commands": ["yarn graphql-codegen --config tools/graphql-codegen.yml"]
      },
      "outputs": ["{projectRoot}/src/lib/generated"]
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/gql-api/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/gql-api"],
      "options": {
        "jestConfig": "libs/gql-api/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["layer:utils", "platform:nodejs"]
}
