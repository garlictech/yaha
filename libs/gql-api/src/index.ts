export * from './lib/clients';
export * from './lib/sdk';
export * from './lib/generated/yaha-api';
