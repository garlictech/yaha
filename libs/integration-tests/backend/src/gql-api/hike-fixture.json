{
  "route": {
    "type": "LineString",
    "coordinates": [
      [19.960138, 47.844277, 510.20901],
      [19.960966, 47.844398, 507.86468],
      [19.961677, 47.844738, 510.82683],
      [19.961677, 47.844738, 510.82683],
      [19.961925, 47.844763, 511.86159],
      [19.96234, 47.844712, 513.6415],
      [19.962839, 47.844414, 514.01338],
      [19.963924, 47.8435, 508.0311],
      [19.964235, 47.843333, 507.4217],
      [19.964432, 47.843173, 505.55983],
      [19.965056, 47.842893, 500.86863],
      [19.966282, 47.842604, 493.08466],
      [19.966765, 47.842604, 490.70485],
      [19.967398, 47.842709, 488.02781],
      [19.967576, 47.842561, 483.8412],
      [19.967755, 47.842505, 481.79085],
      [19.968048, 47.84232, 476.53015],
      [19.968048, 47.84232, 476.53015],
      [19.968521, 47.842177, 472.30818],
      [19.969454, 47.841624, 457.39057],
      [19.969614, 47.841234, 449.36496],
      [19.969851, 47.84118, 446.89692],
      [19.970132, 47.840908, 439.28796],
      [19.970398, 47.840843, 435.73038],
      [19.970712, 47.840468, 424.57916],
      [19.970986, 47.840237, 417.70967],
      [19.971107, 47.840083, 414.42057],
      [19.971095, 47.83993, 412.49989],
      [19.971387, 47.840086, 413.72731],
      [19.971748, 47.840431, 419.11002],
      [19.971982, 47.840508, 420.93228],
      [19.972077, 47.840463, 421.24784],
      [19.97182, 47.840189, 417.03635],
      [19.97182, 47.840189, 417.03635],
      [19.971659, 47.839799, 413.04688],
      [19.97176, 47.839512, 414.50614],
      [19.971828, 47.839088, 415.05464],
      [19.971739, 47.838618, 409.62211],
      [19.971939, 47.837953, 408.9397],
      [19.972784, 47.837068, 411.14911],
      [19.972784, 47.837068, 411.14911],
      [19.97301, 47.837003, 411.03913],
      [19.973218, 47.836832, 408.03474],
      [19.973578, 47.8368, 407.19849],
      [19.973754, 47.837129, 413.89729],
      [19.973937, 47.837323, 417.91868],
      [19.973972, 47.837738, 427.98568],
      [19.974179, 47.838146, 437.43456],
      [19.974179, 47.838146, 437.43456],
      [19.97412, 47.838295, 441.40978],
      [19.974327, 47.838638, 447.34857],
      [19.974582, 47.83888, 449.41355],
      [19.974699, 47.839126, 452.893],
      [19.974943, 47.840176, 464.23136],
      [19.974906, 47.840648, 466.59245],
      [19.975068, 47.840938, 467.3173],
      [19.975326, 47.841212, 468.10353],
      [19.97548, 47.84176, 473.41408],
      [19.975707, 47.84212, 477.28406],
      [19.975801, 47.842364, 480.37765],
      [19.976342, 47.843086, 489.52056],
      [19.976361, 47.843266, 491.28547],
      [19.976672, 47.843572, 496.21222],
      [19.976801, 47.84383, 500.41122],
      [19.977428, 47.84428, 510.93295],
      [19.977608, 47.844459, 515.32195],
      [19.978079, 47.844761, 522.75367],
      [19.978466, 47.844946, 527.38109],
      [19.979244, 47.84562, 538.13192],
      [19.979372, 47.84606, 544.48568],
      [19.979797, 47.846572, 554.01557],
      [19.980119, 47.847246, 567.48612],
      [19.980141, 47.847658, 574.95268],
      [19.980625, 47.848503, 587.22593],
      [19.980769, 47.849029, 590.73452],
      [19.981175, 47.849688, 598.30667],
      [19.98139, 47.850325, 606.753],
      [19.981384, 47.850683, 613.05188],
      [19.981631, 47.851143, 619.53428],
      [19.981612, 47.851547, 625.83157],
      [19.981696, 47.851875, 628.98382],
      [19.981922, 47.852311, 629.66851],
      [19.982273, 47.853501, 633.36545],
      [19.98243, 47.853847, 636.13013],
      [19.982869, 47.854225, 639.18592],
      [19.982964, 47.854439, 643.28085],
      [19.982867, 47.855317, 664.53538],
      [19.983208, 47.856072, 683.03094],
      [19.983208, 47.856072, 683.03094],
      [19.983898, 47.85651, 691.25013],
      [19.9846, 47.856896, 699.28942],
      [19.985406, 47.856979, 697.49616],
      [19.986496, 47.857504, 708.85452],
      [19.986755, 47.857576, 711.70413],
      [19.987478, 47.857999, 724.68808],
      [19.988182, 47.858149, 729.37414],
      [19.988954, 47.858198, 730.44666],
      [19.989408, 47.858373, 735.79325],
      [19.99034, 47.858627, 739.69495],
      [19.99034, 47.858627, 739.69495],
      [19.99057, 47.858787, 744.30098],
      [19.991587, 47.859819, 769.15095],
      [19.991764, 47.860044, 773.20176],
      [19.991856, 47.86042, 779.9734],
      [19.992193, 47.860974, 789.86979],
      [19.992439, 47.861579, 800.87295],
      [19.992619, 47.861775, 803.30857],
      [19.992816, 47.862138, 805.71179],
      [19.992997, 47.862361, 807.56196],
      [19.993482, 47.862702, 812.35373],
      [19.994007, 47.862956, 817.94407],
      [19.994263, 47.863376, 821.46987],
      [19.994714, 47.864351, 829.79796],
      [19.994703, 47.86469, 829.93499],
      [19.995813, 47.865429, 845.78666],
      [19.995813, 47.865429, 845.78666],
      [19.996571, 47.86567, 855.57793],
      [19.996794, 47.865813, 856.70046],
      [19.997301, 47.866035, 859.39548],
      [19.998312, 47.866259, 867.96286],
      [19.998858, 47.866479, 872.97125],
      [19.999401, 47.866447, 878.34803],
      [20.000425, 47.866631, 886.61491],
      [20.000656, 47.866714, 889.07936],
      [20.000884, 47.866994, 892.15904],
      [20.001352, 47.867117, 898.23356],
      [20.001594, 47.867275, 902.13586],
      [20.00184, 47.86736, 904.08126],
      [20.002183, 47.867387, 903.87522],
      [20.002272, 47.867446, 904.48967],
      [20.002206, 47.867684, 907.96993],
      [20.002226, 47.867828, 909.17417],
      [20.002652, 47.868157, 912.11857],
      [20.002652, 47.868157, 912.11857],
      [20.002983, 47.868307, 914.40459],
      [20.003101, 47.868421, 916.82359],
      [20.003302, 47.868494, 918.80483],
      [20.003631, 47.869047, 931.77631],
      [20.003664, 47.869584, 941.52043],
      [20.003787, 47.869662, 943.45543],
      [20.004219, 47.870142, 955.07486],
      [20.004578, 47.870305, 961.41449],
      [20.004971, 47.870329, 965.53891],
      [20.005404, 47.870444, 972.58584],
      [20.005698, 47.870625, 979.94516],
      [20.006021, 47.870734, 984.06115],
      [20.006468, 47.871013, 987.56142],
      [20.006553, 47.871105, 988.33389],
      [20.006318, 47.871506, 989.7454],
      [20.006251, 47.871952, 993.05262],
      [20.006276, 47.8721, 994.78106],
      [20.006468, 47.872289, 998.98767],
      [20.006636, 47.872407, 1002.44973],
      [20.008136, 47.872489, 1013.59136],
      [20.008222, 47.872716, 1014.73289],
      [20.008467, 47.87298, 1016.56211],
      [20.008222, 47.872716, 1014.73289],
      [20.008136, 47.872489, 1013.59136],
      [20.007261, 47.872453, 1008.84573],
      [20.006636, 47.872407, 1002.44973],
      [20.006636, 47.872407, 1002.44973],
      [20.006556, 47.872401, 1001.45624],
      [20.006359, 47.872383, 998.80676],
      [20.00606, 47.872398, 995.15118],
      [20.003706, 47.872322, 972.40201],
      [20.000165, 47.871996, 931.18687],
      [19.998669, 47.871937, 913.66032],
      [19.997435, 47.871964, 902.28433],
      [19.995971, 47.872145, 890.10688],
      [19.995171, 47.87221, 883.02672],
      [19.994103, 47.872204, 874.65992],
      [19.993734, 47.872174, 871.1699],
      [19.991757, 47.87226, 845.52638],
      [19.990115, 47.872234, 829.18604],
      [19.988753, 47.872374, 819.763],
      [19.986992, 47.872412, 794.12066],
      [19.985118, 47.872496, 777.43452],
      [19.985118, 47.872496, 777.43452],
      [19.984552, 47.872461, 771.5909],
      [19.984176, 47.872394, 767.58573],
      [19.983807, 47.872255, 762.55293],
      [19.983357, 47.871747, 757.7791],
      [19.983046, 47.87156, 756.32336],
      [19.982689, 47.871541, 754.15197],
      [19.982512, 47.871477, 753.67028],
      [19.982113, 47.871233, 752.38422],
      [19.981162, 47.870424, 739.56924],
      [19.981084, 47.87028, 738.16672],
      [19.981101, 47.870091, 737.41099],
      [19.981352, 47.869278, 727.48791],
      [19.981352, 47.869278, 727.48791],
      [19.981329, 47.86917, 725.84047],
      [19.981187, 47.869021, 723.83243],
      [19.981072, 47.868974, 723.30544],
      [19.980818, 47.868959, 723.08375],
      [19.980192, 47.86924, 725.70329],
      [19.979109, 47.869354, 720.14781],
      [19.978647, 47.869434, 716.3065],
      [19.978367, 47.868979, 714.90828],
      [19.978241, 47.86901, 714.78854],
      [19.978038, 47.868593, 714.45605],
      [19.977712, 47.868714, 716.08751],
      [19.977418, 47.868732, 717.71108],
      [19.976609, 47.868465, 722.24395],
      [19.975919, 47.868005, 721.39436],
      [19.975577, 47.86784, 720.81771],
      [19.975214, 47.867754, 721.55264],
      [19.975214, 47.867754, 721.55264],
      [19.97479, 47.867328, 717.21638],
      [19.974448, 47.867136, 715.33865],
      [19.97413, 47.867028, 714.83317],
      [19.973913, 47.866773, 710.51943],
      [19.973643, 47.866635, 709.33052],
      [19.973415, 47.86659, 709.37717],
      [19.972502, 47.866711, 710.66647],
      [19.971284, 47.866458, 695.74329],
      [19.971055, 47.866226, 692.95442],
      [19.970915, 47.865815, 689.71884],
      [19.969763, 47.865012, 680.74906],
      [19.969247, 47.864178, 672.99488],
      [19.969167, 47.863895, 671.00914],
      [19.968238, 47.863118, 671.45647],
      [19.967887, 47.862732, 669.02704],
      [19.967581, 47.862516, 668.68124],
      [19.967012, 47.862226, 665.77614],
      [19.965673, 47.861411, 653.58082],
      [19.964544, 47.860659, 642.97639],
      [19.964544, 47.860659, 642.97639],
      [19.964042, 47.860382, 641.18735],
      [19.963946, 47.860072, 638.3371],
      [19.963243, 47.859174, 633.89459],
      [19.962868, 47.858389, 631.59873],
      [19.962512, 47.857844, 629.00565],
      [19.962393, 47.857308, 621.08628],
      [19.961972, 47.856439, 606.70519],
      [19.961913, 47.855999, 600.38608],
      [19.96174, 47.855743, 597.86621],
      [19.961384, 47.855445, 594.92796],
      [19.960542, 47.854926, 587.1622],
      [19.960083, 47.854509, 578.34155],
      [19.95904, 47.853873, 559.57845],
      [19.95879, 47.853618, 554.78413],
      [19.958107, 47.853059, 544.53496],
      [19.957891, 47.852828, 539.36488],
      [19.957677, 47.852264, 534.56307],
      [19.957078, 47.851493, 528.69876],
      [19.95652, 47.851092, 521.28961],
      [19.956036, 47.85045, 517.19881],
      [19.956036, 47.85045, 517.19881],
      [19.955617, 47.850148, 513.33445],
      [19.955501, 47.849886, 513.51388],
      [19.955449, 47.849533, 516.46725],
      [19.955262, 47.849151, 518.18253],
      [19.955051, 47.848476, 521.7843],
      [19.955214, 47.847872, 523.93814],
      [19.955032, 47.847585, 522.24326],
      [19.955004, 47.847222, 521.56231],
      [19.955086, 47.846278, 518.14657],
      [19.955956, 47.84554, 513.82621],
      [19.955956, 47.84554, 513.82621],
      [19.957595, 47.84453, 512.10618],
      [19.958117, 47.8445, 512.86974],
      [19.959652, 47.844274, 511.7087],
      [19.959725, 47.844238, 511.29579],
      [19.96012, 47.844275, 510.26954]
    ]
  },
  "description": [
    {
      "languageKey": "hu_HU",
      "title": "Sástó–Kékes körtúra",
      "summary": "Télen sem érdemes beszorulni a négy fal közé. A bevitt kalóriák elégetésének kevés élvezetesebb módja van, mint a behavazott, csendes erdőben felsétálni hazánk legmagasabb csúcsára, és gyönyörködni a vakítóan fehér panorámában. ",
      "type": "markdown"
    }
  ]
}
