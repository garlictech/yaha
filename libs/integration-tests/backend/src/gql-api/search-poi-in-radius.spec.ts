import { getGraphqlSdkForIAM, YahaApi } from '@yaha/gql-api';
import { delay, forkJoin, switchMap, tap } from 'rxjs';
import * as R from 'ramda';

const poiId = '12-search-poi-in-radius-c86e6484-a2c7-11ec-b909-0242ac120002';

const sdk = getGraphqlSdkForIAM(
  process.env.AWS_ACCESS_KEY_ID || '',
  process.env.AWS_SECRET_ACCESS_KEY || '',
);

const cleanup$ = forkJoin([sdk.DeletePoi({ input: { id: poiId } })]);

afterAll(done => {
  cleanup$.subscribe(() => done());
});

test('Search for a poi in radius', done => {
  sdk
    .CreatePoi({
      input: {
        id: poiId,
        location: {
          lat: 1,
          lon: 1,
        },
      },
    })
    .pipe(
      delay(3000),
      switchMap(() =>
        sdk.SearchByRadius({
          query: {
            location: { lat: 1, lon: 1 },
            radiusInMeters: 100,
            objectType: YahaApi.GeoSearchableObjectType.poi,
            limit: 1,
          },
        }),
      ),
      tap(console.warn),
      tap(res =>
        expect(R.omit(['nextToken'], res)).toMatchSnapshot('SHOULD FIND poi'),
      ),
      switchMap(() =>
        sdk.SearchByRadius({
          query: {
            location: { lat: 2, lon: 2 },
            radiusInMeters: 100,
            objectType: YahaApi.GeoSearchableObjectType.poi,
            limit: 1,
          },
        }),
      ),
      tap(console.warn),
      tap(res =>
        expect(R.omit(['nextToken'], res)).toMatchSnapshot(
          'SHOULD NOT FIND poi',
        ),
      ),
    )
    .subscribe(() => done());
}, 10000);
