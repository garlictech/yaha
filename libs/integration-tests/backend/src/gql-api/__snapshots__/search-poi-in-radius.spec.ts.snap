// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search for a poi in radius: SHOULD FIND poi 1`] = `
Object {
  "items": Array [
    "12-search-poi-in-radius-c86e6484-a2c7-11ec-b909-0242ac120002",
  ],
  "total": 1,
}
`;

exports[`Search for a poi in radius: SHOULD NOT FIND poi 1`] = `
Object {
  "items": Array [],
  "total": 0,
}
`;
